src/rsz/src/SizeDownMove.cc

```diff
@@ -172,171 +168,297 @@ bool SizeDownMove::doMove(const Path* drvr_path,
+      // using simple RC model. This is specifically so that we can downsize
+      // the critical fanout if it gets a net improvement in delay.
+      float drvr_res = drvr_port->driveResistance();
+      float drvr_delta_delay = -drvr_res * (load_input_cap - new_input_cap);
+
+      // Find the gate delay with the new gate and old output cap
+      float new_load_delay
+          = resizer_->gateDelay(output_port, output_caps[i], dcalc_ap);
 
-      float new_drvr_delay
-          = resizer_->gateDelay(drvr_port, new_drvr_load_cap, dcalc_ap);
-      float new_fanout_delay = resizer_->gateDelay(
-          new_fanout_output_port, fanout_load_cap, dcalc_ap);
-      debugPrint(
-          logger_,
-          RSZ,
-          "size_down",
-          3,
-          " new delay {} FO={} drvr_delay {}<{}, fanout_delay: {}<{} ({}+{}) ",
-          network_->pathName(fanout_pin),
-          swappable->name(),
-          delayAsString(new_drvr_delay, sta_, 3),
-          delayAsString(drvr_delay, sta_, 3),
-          delayAsString(new_fanout_delay, sta_, 3),
-          delayAsString(fanout_delay - fanout_slack, sta_, 3),
-          delayAsString(fanout_delay, sta_, 3),
-          delayAsString(fanout_slack, sta_, 3));
-      // Check if the combined new delay is better than the old one
-      // If fanout slack is positive, this allows us to slow down the fanout
-      // gate If fnaout slack is negative, this allows us to accept it as long
-      // as the total delay improves
-      if (new_drvr_delay + new_fanout_delay
-          < drvr_delay + fanout_delay + fanout_slack) {
-        best_cell = swappable;
-        best_cap = new_fanout_cap;
-        best_area = new_area;
+      // This includes the improvement in delay of the driver gate
+      float new_total_delay = new_load_delay + drvr_delta_delay;
+
+      float delay_change = new_total_delay - output_delays[i];
+
+      debugPrint(logger_,
+                 RSZ,
+                 "size_down",
+                 4,
+                 " new delay {}->{} gate={} drvr_delta {} + new_delay {} - "
+                 "old_delay {} < "
+                 "slack {} ({} < {})",
+                 network_->pathName(load_pin),
+                 network_->pathName(output_pins[i]),
+                 swappable->name(),
+                 delayAsString(drvr_delta_delay, sta_, 3),
+                 delayAsString(new_load_delay, sta_, 3),
+                 delayAsString(output_delays[i], sta_, 3),
+                 delayAsString(slack_margin, sta_, 3),
+                 delayAsString(delay_change, sta_, 3),
+                 delayAsString(slack_margin, sta_, 3));
+
+      // First case is positive slack and delay change doesn't get worse than
+      // that slack. Second case is negative slack and delay is improved, but
+      // doesn't necessarily fix the slack violation entirely.
+      if ((slack_margin > 0 && delay_change > slack_margin)
+          || (slack_margin < 0 && delay_change > 0)) {
```

Comment:
We don't check paths through the gate which don't go through `load_pin`; also for flops this check will be comparing delay in driving `Q` with slack on `D`.

Reply:
> We don't have to check other paths that aren't being sized down,

Oops, I'm not being clear. I mean let's say the gate we are downsizing is a NAND, we only check the slack on pin A but there could be a critical path going through pin B which we don't check for. One way to include paths going through B would be to build the check off the slack on the output pin of the NAND gate, but then it's not precise to include `drvr_delta_delay` for paths going through B.

Reply:
Got it. Yes, you are correct.